src/torchmetrics/functional/classification/precision_recall.py

```diff
@@ -56,7 +57,9 @@ def _precision_recall_reduce(
         return _safe_divide(tp, tp + different_stat, zero_division)
 
     score = _safe_divide(tp, tp + different_stat, zero_division)
-    return _adjust_weights_safe_divide(score, average, multilabel, tp, fp, fn, top_k=top_k)
+    return _adjust_weights_safe_divide(
+        score=score, average=average, multilabel=multilabel, tp=tp, fp=fp, fn=fn, ignore_index=ignore_index, top_k=top_k
```

Comment:
@rittik9 mind checking what is wrong here, seems that the computed metrics do not match the reference...